test/exprtest/blas1_axpy_copy_test.cpp

```diff
@@ -0,0 +1,133 @@
+/***************************************************************************
+ *
+ *  @license
+ *  Copyright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename blas1_axpy_copy_test.cpp
+ *
+ **************************************************************************/
+#include "blas_test.hpp"
+#include "sycl_blas.hpp"
+typedef ::testing::Types<blas_test_float<>, blas_test_double<> > BlasTypes;
+
+TYPED_TEST_CASE(BLAS_Test, BlasTypes);
+
+REGISTER_SIZE(::RANDOM_SIZE, axpy_copy_test)
+REGISTER_STRD(::RANDOM_STRD, axpy_copy_test)
+REGISTER_PREC(float, 1e-4, axpy_copy_test)
+REGISTER_PREC(double, 1e-6, axpy_copy_test)
+REGISTER_PREC(std::complex<float>, 1e-4, axpy_copy_test)
+REGISTER_PREC(std::complex<double>, 1e-6, axpy_copy_test)
+
+TYPED_TEST(BLAS_Test, axpy_copy_test) {
+  using scalar_t = typename TypeParam::scalar_t;
+  using ExecutorType = typename TypeParam::executor_t;
+  using TestClass = BLAS_Test<TypeParam>;
+  using test = class axpy_copy_test;
+
+  int size = TestClass::template test_size<test>();
+  int strd = TestClass::template test_strd<test>();
+  scalar_t prec = TestClass::template test_prec<test>();
+
+  DEBUG_PRINT(std::cout << "size == " << size << std::endl);
+  DEBUG_PRINT(std::cout << "strd == " << strd << std::endl);
+
+  // setting alpha to some value
+  scalar_t alpha(1.54);
+  
+  // vector X will be copied to Y
+  std::vector<scalar_t> vX(size, 0);
+  std::vector<scalar_t> vY(size, 0);
+
+  // vector Z will be copied to W
+  std::vector<scalar_t> vZ(size, 0);
+  std::vector<scalar_t> vW(size, 0);
+
+  // vectors Y and W will be used in the AXPY operation
+
+  TestClass::set_rand(vX, size);
+  TestClass::set_rand(vZ, size);
+
+  // vector A will store the result of AXPY for comparison later
+  std::vector<scalar_t> vA(size, 0);
+
+
+  // First copy X->Y, and Z->W
+  for (int i = 0; i < size; ++i) {
```

Comment:
Is the stride used for the copy? if so this will be incorrect

Reply:
New test code should be correct.